activemq-client/src/main/java/org/apache/activemq/transport/nio/NIOTransport.java

```diff
@@ -139,8 +139,10 @@ protected void serviceRead() {
                     nextFrameSize = inputBuffer.getInt() + 4;
 
                     if (wireFormat instanceof OpenWireFormat) {
-                        long maxFrameSize = ((OpenWireFormat)wireFormat).getMaxFrameSize();
-                        if (nextFrameSize > maxFrameSize) {
+                        OpenWireFormat openWireFormat = (OpenWireFormat)wireFormat;
+                        long maxFrameSize = openWireFormat.getMaxFrameSize();
+
+                        if (openWireFormat.isMaxFrameSizeEnabled() && nextFrameSize > maxFrameSize) {
                             throw new IOException("Frame size of " + (nextFrameSize / (1024 * 1024)) + " MB larger than max allowed " + (maxFrameSize / (1024 * 1024)) + " MB");
```

Comment:
I'd use the createFrameSizeException() here

Reply:
Let's do that part in 5.17.x since it could (low change) break server-side log parsers

ref: https://issues.apache.org/jira/browse/AMQ-8476